Framework/DataHandling/test/LoadSampleShapeTest.h

```diff
@@ -0,0 +1,211 @@
+#ifndef LOAD_SHAPETEST_H_
+#define LOAD_SHAPETEST_H_
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Sample.h"
+#include "MantidDataHandling/LoadSampleShape.h"
+#include "MantidDataHandling/LoadInstrument.h"
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/OptionalBool.h"
+#include "MantidTestHelpers/ComponentCreationHelper.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::DataHandling;
+using namespace Mantid::Geometry;
+
+class LoadSampleShapeTest : public CxxTest::TestSuite {
+public:
+  static LoadSampleShapeTest *createSuite() {
+    return new LoadSampleShapeTest();
+  }
+  static void destroySuite(LoadSampleShapeTest *suite) { delete suite; }
+
+  void testName() { TS_ASSERT_EQUALS(loadShape.name(), "LoadSampleShape"); }
+
+  void testVersion() { TS_ASSERT_EQUALS(loadShape.version(), 1); }
+
+  void testInit() {
+
+    TS_ASSERT_THROWS_NOTHING(loadShape.initialize());
+    TS_ASSERT(loadShape.isInitialized());
+
+    TSM_ASSERT_EQUALS("should be 3 properties here", 3,
+                      (size_t)(loadShape.getProperties().size()));
+  }
+
+  void testConfidence() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    std::string path = testLoadSampleShape.getPropertyValue("Filename");
+    auto *descriptor = new Kernel::FileDescriptor(path);
+    TS_ASSERT_EQUALS(90, testLoadSampleShape.confidence(*descriptor));
+    delete descriptor;
+  }
+
+  void testExec_2WS() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    prepareWorkspaces(testLoadSampleShape, "InputWS", "OutputWS");
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
+    clearWorkspaces("InputWS", "OutputWS");
+  }
+
+  void testExec_1WS() {
+    LoadSampleShape testLoadSampleShape;
+    testLoadSampleShape.initialize();
+    testLoadSampleShape.setPropertyValue("Filename", "cube.stl");
+    prepareWorkspaces(testLoadSampleShape, "InputWS", "InputWS");
+    TS_ASSERT_THROWS_NOTHING(testLoadSampleShape.execute());
+    TS_ASSERT(testLoadSampleShape.isExecuted());
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_output_workspace_has_MeshObject_2WS() {
+    loadMeshObject("InputWS", "OutputWS", "cube.stl");
+    clearWorkspaces("InputWS", "OutputWS");
+  }
+
+  void test_output_workspace_has_MeshObject_1WS() {
+    loadMeshObject("InputWS", "InputWS", "cube.stl");
+    clearWorkspaces("InputWS", "InputWS");
+  }
+
+  void test_cube() {
+    auto cube = loadMeshObject("InputWS", "InputWS", "cube.stl");
```

Comment:
Rather than test the `Algorithm` as one big block, did you consider breaking it up into testable functions/classes etc. The reason I ask is that you are only interested in the `Sample` but for your testing (since you do that via the `Algorithm`), you have to mess around with getting things in and out of `Workspaces`. This is just a consideration you should think about.